django/tasks/backends.py

```diff
@@ -0,0 +1,119 @@
+from . import constants
+from .base import get_backend
+from .exceptions import ResultUnavailable
+
+
+class TaskResult(object):
+    """
+    Wrapper class representing a status of a task. Provides methods to retrieve
+    the task's status and result value/error.
+    """
+    def __init__(self, backend, task_id):
+        if not isinstance(backend, BaseBackend):
```

Comment:
I don't think there's a good reason to enforce types like this. How about replacing this with `if isinstance(backend, basestring)`?